library/test/src/cli.rs

```diff
@@ -446,16 +449,22 @@ fn get_color_config(matches: &getopts::Matches) -> OptPartRes<ColorConfig> {
     Ok(color)
 }
 
-fn get_nocapture(matches: &getopts::Matches) -> OptPartRes<bool> {
-    let mut nocapture = matches.opt_present("nocapture");
-    if !nocapture {
-        nocapture = match env::var("RUST_TEST_NOCAPTURE") {
+fn get_capture(matches: &getopts::Matches) -> OptPartRes<bool> {
+    let no_capture = matches.opt_present("no-capture");
+    let nocapture = matches.opt_present("nocapture");
+    if nocapture {
+        let _ = write!(io::stderr(), "warning: `--nocapture` is deprecated, use `--no-capture`");
```

Comment:
This makes it impossible to have a command support both latest Rust and one or two Rusts back without a warning. In the compiler we always try to avoid such situations, I think libtest should do the same. Add the new flag now, the deprecate the old one in two versions or something like that.